test/unit/resources/aws_dynamodb_table_test.rb

```diff
@@ -0,0 +1,94 @@
+require 'helper'
+require 'aws_dynamodb_table'
+require 'aws-sdk-core'
+
+class AwsDynamodbTableConstructorTest < Minitest::Test
+
+  def test_empty_params_not_ok
+    assert_raises(ArgumentError) { AwsDynamodbTable.new(client_args: { stub_responses: true }) }
+  end
+
+  def test_accepts_valid_args
+    AwsDynamodbTable.new(table_name: 'valid_table_name', client_args: { stub_responses: true })
+  end
+
+  def test_rejects_unrecognized_params
+    assert_raises(ArgumentError) { AwsDynamodbTable.new(rubbish: 9) }
+  end
+
+  def test_fails_with_more_than_one_parameter
+    assert_raises(ArgumentError) { AwsDynamodbTable.new(table_name: 'fail with more than 1 paramenter', table_arn: 'aws_arn') }
+  end
+
+  def test_table_non_existing
+    refute AwsDynamodbTable.new(table_name: 'table_that_doesnt_exist', client_args: { stub_responses: false }).exists?
+  end
+end
+
+class AwsDynamodbTableTest < Minitest::Test
+  def setup
+    mock_table = {}
```

Comment:
There are methods in `mock/aws_givens.rb` which will generate random test data for most of these fields, so the test doesn't use static hard coded values.